.github/actions/static-analysis-tests/action.yml

```diff
@@ -0,0 +1,52 @@
+name: Terraform Static Analysis Tests
+description: Runs Terraform Static Analysis Tests - format, validate and plan
+inputs:
+  GITHUB_TOKEN:
+    description: Workflow GitHub Token (limited permissions)
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Terraform Validate and Plan
+      shell: bash
+      run: |
+        terraform init
+        terraform fmt --recursive
+        terraform validate
+        terraform plan -lock=false -out terraform.plan
+        terraform show -no-color terraform.plan > terraform.text
+
+    - name: Report Terraform Plan
+      uses: actions/github-script@v6
+      with:
+        github-token: ${{ inputs.GITHUB_TOKEN }}
```

Comment:
🔧  Internally, this is set to `github.token` which has the same value as `secrets.GITHUB_TOKEN`. Unless you see a situation where this would be changed to another token, this can be removed.

```suggestion
```

Reply:
removed!